src/shared/openssl-util.h

```diff
@@ -54,9 +53,7 @@ DEFINE_TRIVIAL_CLEANUP_FUNC_FULL(SSL*, SSL_free, NULL);
 DEFINE_TRIVIAL_CLEANUP_FUNC_FULL(BIO*, BIO_free, NULL);
 DEFINE_TRIVIAL_CLEANUP_FUNC_FULL(EVP_MD_CTX*, EVP_MD_CTX_free, NULL);
 DEFINE_TRIVIAL_CLEANUP_FUNC_FULL(ASN1_OCTET_STRING*, ASN1_OCTET_STRING_free, NULL);
-DISABLE_WARNING_DEPRECATED_DECLARATIONS;
-DEFINE_TRIVIAL_CLEANUP_FUNC_FULL(ENGINE*, ENGINE_free, NULL);
```

Comment:
It was explicitly requested to have this definition in the header when it was added, so leave this here

Reply:
i think you mean a comment from @poettering ? aslong as no user is expected to need it I would leave it there.

Reply:
Yes, it was asked to have it here regardless of whether it's used outside of this module, so please leave it here

Reply:
well, lets ping @poettering, if this specific instance is really an issue?
it seemed more like a question to me.

Reply:
Link to comment is here: https://github.com/systemd/systemd/pull/31261#discussion_r1484311058

The definition is only (to be) used via `openssl_load_key_from_token`, is conditional and deprecated,should not end up in the header-

Reply:
Even it is requested in the linked comment, but that's not necessary in the header, and keeping the functions in header requires further ifdef and include in the header. That's annoying. I agree with @nolange.

Reply:
yeah, with the ifdeffery might make more sense to move this specific case if it's really not needed anywhere else, and is unlikely to be.